classes/Cart.php

```diff
@@ -1628,15 +1628,9 @@ public function _addCustomization($id_product, $id_product_attribute, $index, $t
      */
     public function orderExists()
     {
-        $cache_id = 'Cart::orderExists_' . (int) $this->id;
-        if (!Cache::isStored($cache_id)) {
-            $result = (bool) Db::getInstance()->getValue('SELECT count(*) FROM `' . _DB_PREFIX_ . 'orders` WHERE `id_cart` = ' . (int) $this->id);
-            Cache::store($cache_id, $result);
+        $result = (bool) Db::getInstance()->getValue('SELECT count(*) FROM `' . _DB_PREFIX_ . 'orders` WHERE `id_cart` = ' . (int) $this->id);
 
-            return $result;
-        }
-
-        return Cache::retrieve($cache_id);
+        return $result;
     }
```

Comment:
The cache only makes sense if it's invalidated properly. If an order is checked, then deleted, then checked again it would seem as if it wasn't deleted. So yeah, I think it's correct. However I'm interested in knowing why it was added in the first place.

Reply:
The ugly reason: https://github.com/PrestaShop/PrestaShop/commit/a7869a1f06c32978def39d4ab9ff9b2eebe14344

Reply:
👍